pdf_craft/transformer/xml_translator/xml/xml_like.py

```
@@ -0,0 +1,279 @@
+    def __init__(self, file: IO[bytes], is_html_like: bool = False) -> None:
+        raw_content = file.read()
+        self._is_html_like = is_html_like
+        self._encoding: str = self._detect_encoding(raw_content)
+        content = raw_content.decode(self._encoding)
+        self._header, xml_content = self._extract_header(content)
+        self._namespaces: dict[str, str] = {}
+        self._tag_to_namespace: dict[str, str] = {}
+        self._attr_to_namespace: dict[str, str] = {}
+
+        try:
+            # 不必判断类型，这是一个防御性极强的函数，可做到 shit >> XML
```

Ruff (0.16.1)

[warning] Comment contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF003)